packages/notifications/push/amplify_push_notifications/lib/src/amplify_push_notifications_impl.dart

```diff
@@ -90,22 +92,40 @@ class AmplifyPushNotifications extends PushNotificationsPluginInterface {
   }
 
   @override
-  Stream<String> get onTokenReceived => _onTokenReceived;
+  Stream<String> get onTokenReceived {
+    if (!_isConfigured) {
+      throw const PushNotificationException(
+        'Configure Amplify with Notifications Plugin before adding the listener.',
```

Comment:
nit: If plan to use the same error message (it the API context doesn't really matter), can make this exception a top level constant.